src/components/ReservationForm.tsx

```diff
@@ -0,0 +1,490 @@
+import { CalendarUIProps, CalendarAction } from "../calendar/types";
+import {
+  IconButton,
+  makeStyles,
+  Dialog,
+  Toolbar,
+  Typography,
+  DialogContent,
+  Button,
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import Select from "./Select";
+import { AuthContext } from "./AuthContext";
+import { makeTransition } from "./Transition";
+import UserGroup from "../resources/UserGroup";
+import { Formik } from "formik";
+import * as Yup from "yup";
+import GearForm from "./GearForm";
+import QuantityList from "./QuantityList";
+import { ResourceKey } from "../resources/types";
+import Project from "../resources/Project";
+import Equipment from "../resources/Equipment";
+
+const useStyles = makeStyles(() => ({
+  guests: {
+    display: "none",
+  },
+  notes: {
+    display: "none",
+  },
+  gearList: {
+    display: "none",
+  },
+  list: {
+    display: "flex",
+    justifyContent: "flex-start",
+    alignContent: "row",
+  },
+}));
+
+const transition = makeTransition("left");
+
+const ReservationForm: FunctionComponent<CalendarUIProps> = ({
+  dispatch,
+  state,
+}) => {
+  // Get values from App
+  const { user } = useContext(AuthContext);
+  const groups = state.resources[ResourceKey.Groups] as UserGroup[];
+  const projects = state.resources[ResourceKey.Projects] as Project[];
+  const gear = state.resources[ResourceKey.Equipment] as Equipment[];
+
+  console.log(gear);
+  // Function to convert Gear Array to Quantized Gear Array
+  function quantizeGear(gear: Equipment[]): Equipment[] {
+    const tempArray: Equipment[] = [];
+    gear.forEach((item) => {
+      item.quantity = gear.filter(
+        (element) => element.title === item.title
+      ).length;
+      const index = tempArray.findIndex(
+        (element) => element.title === item.title
+      );
+      if (index === -1) {
+        tempArray.push(item);
+      }
+    });
+    return tempArray;
+  }
+  //const gear: Equipment[] = quantizeGear(state.gear);
+
+  // Constant Declatations
+  const initialGroups: UserGroup[] = [];
+  const classes = useStyles();
+  const phoneRegExp = /^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/;
+  const initialValidationSchema = Yup.object().shape({
+    phone: Yup.string().matches(phoneRegExp, "Phone number is not valid"),
+    description: Yup.string().required("Required"),
+  });
+  const filters: { [k: string]: boolean } = {};
+  const categories: { [k: string]: Set<string> } = {};
+  const quantities: { [k: string]: number } = {};
+
+  // Build quantities dictionary for Formik
+  // Build Categories Dictionary
+  // Build Filters Dictionary
+  gear.forEach((item) => {
+    quantities[item.description] = 0;
+    item.tags.forEach((tag) => {
+      if (!categories[item.category]) {
+        categories[item.category] = new Set();
+      }
+      categories[item.category].add(tag.name);
+      filters[tag.name] = false;
+    });
+  });
```

Comment:
typescript shouting about the undeclared types of categories, filters, and quantities in that example, I'll need to spend more time on it. Before starting this change, I had `// Function to convert equipment Array to Quantized equipment Array
export function buildDictionaries(
  equipment: Equipment[]
): [
  { [k: string]: boolean },
  { [k: string]: Set<string> },
  { [k: string]: number }
] {
  // Build quantities dictionary for Formik
  // Build Categories Dictionary
  // Build Filters Dictionary
  const filters: { [k: string]: boolean } = {};
  const categories: { [k: string]: Set<string> } = {};
  const quantities: { [k: string]: number } = {};
  equipment.forEach((item) => {
    quantities[item.description] = 0;
    item.tags.forEach((tag) => {
      if (!categories[item.category]) {
        categories[item.category] = new Set();
      }
      categories[item.category].add(tag.name);
      filters[tag.name] = false;
    });
  });
  return [filters, categories, quantities];
}
` and I appreciate the need to change from forEach to reduce

Reply:
Sorry, I was just using the suggestion tool as a way to highlight the code, not really meaning for you to use GitHub to drop it in.  I will stop abusing the suggestion tool in that way!

I made this suggestion again, with typings, in the new file.  Check there, and I would advise copying pasting the code into your editor rather than using the auto commit github thing.